libtransmission/torrent.h

```diff
@@ -711,39 +711,114 @@ struct tr_torrent final : public tr_completion::torrent_view
         return bandwidth_group_;
     }
 
+    [[nodiscard]] constexpr auto peer_limit() const noexcept
+    {
+        return max_connected_peers_;
+    }
+
+    // --- idleness
+
+    constexpr void set_idle_limit_mode(tr_idlelimit mode) noexcept
+    {
+        if (mode == TR_IDLELIMIT_GLOBAL || mode == TR_IDLELIMIT_SINGLE || mode == TR_IDLELIMIT_UNLIMITED)
```

Comment:
Maybe adding back an assert is a good idea, we don't want to write bugs and have libtransmission silently ignore them for us.

```cpp
TR_ASSERT(mode == TR_IDLELIMIT_GLOBAL || mode == TR_IDLELIMIT_SINGLE || mode == TR_IDLELIMIT_UNLIMITED);
```